cli/pcluster/config/pcluster_config.py

```diff
@@ -157,21 +166,37 @@ def add_section(self, section):
 
         :param section, a Section object
         """
-        if section.key not in self.sections:
-            self.sections[section.key] = {}
+        if section.key not in self.__sections:
+            self.__sections[section.key] = {}
 
         section_label = section.label if section.label else section.definition.get("default_label", "default")
-        self.sections[section.key][section_label] = section
+        self.__sections[section.key][section_label] = section
+        # Sections structure is altered; mark for refresh
+        self.refresh()
 
-    def remove_section(self, section_key, section_label):
+    def remove_section(self, section_key, section_label=None):
         """
         Remove a section from the PclusterConfig object, if there.
 
         :param section_key: the identifier of the section type
         :param section_label: the label of the section to delete.
         """
-        if section_key in self.sections:
-            self.sections[section_key].pop(section_label, None)
+        if section_key in self.__sections:
+            sections = self.__sections[section_key]
+
+            if section_label:
+                # If section label is specified, remove it directly
+                if section_key in self.__sections:
+                    sections.pop(section_label, None)
```

Comment:
Yes because in one case we are removing the item from the dict of sections with the same key while in the other we are removing the entire dict corresponding to that key.

Reply:
Got it, I think the variable names sections vs __sections doesn't help to understand the difference